assets/src/application/services/subscription/useSubscriptionUIRegistry.ts

```diff
@@ -0,0 +1,69 @@
+import { useState, useEffect } from 'react';
+import { assocPath, dissocPath, omit, pathOr } from 'ramda';
+import { AnyObject } from '@appServices/utilities/types';
+
+import { ElementRegistry, SubscriptionUIRegistry, SubscriptionUIRegistryHook } from './types';
+
+type SUIRhook = SubscriptionUIRegistryHook;
+type SUIR = SubscriptionUIRegistry;
+
+let elementRegistry: ElementRegistry = {};
+
+const useSubscriptionUIRegistry: SUIRhook = ({ domain, service, path }) => {
```

Comment:
i wonder if it would help simplify things if we try to remove as many non-critical terms from names as much as possible.

SO for example this could be simply `useUIRegistry` or `useElementRegistry`

Reply:
great suggestion, we can take advantage of the folder namespacing